doc/architecture-specification.md

```diff
@@ -0,0 +1,72 @@
+# emojicoin dot fun Architecture Specification
+
+The key words `MUST`, `MUST NOT`, `REQUIRED`, `SHALL`, `SHALL NOT`, `SHOULD`,
+`SHOULD NOT`, `RECOMMENDED`,  `MAY`, and `OPTIONAL` in this document are to be
+interpreted as described in [RFC 2119].
+
+These keywords `SHALL` be in `monospace` for ease of identification.
+
+## General mechanism
+
+1. Each emojicoin market `SHALL` be denominated in `APT`.
+1. Each market `SHALL` operate in two states: an initial bonding curve, then a
+   simple constant product pool.
+1. The bonding curve state `SHALL` implement a concentrated liquidity automated
+   market maker curve that contains purely emojicoins upon initialization,
+   eventually exhausted fully into `APT` upon bonding curve state completion.
+1. After completing the bonding curve state, the `APT` generated from initial
+   buy orders `SHALL` be locked into a constant product pool with set aside
+   emojicoin reserves, such that market price is identical before and after the
+   state transition.
+1. The entire emojicoin supply `SHALL` be minted upon market initialization,
+   with mint, burn, and freeze capabilities destroyed after minting total
+   supply, to be allocated for bonding curve and constant-product states.
+1. A buy order that breaks the market out of the bonding curve `SHALL` continue
+   to fill after the state transition, rather than fragmenting the order into
+   two transactions.
+
+## Indexing
+
+1. The user interface `SHALL` rely solely on the Aptos GraphQL endpoint for
+   indexing purposes, enabling for example a leaderboard based on market
+   capitalization of each emojicoin market.
+
+## Frontend
+
+1. The frontend `SHALL` implement a market capitalization leaderboard with
+   logarithmic to enable viewing of prevalent markets.
+1. The frontend `SHALL` enable a king-of-the-hill style mechanism to show
+   emojicoins passing others in market capitalization.
+1. The frontend `SHALL` enable permissionless registration of a new emojicoin
+   market that has not yet been registered.
+
+## Coin definitions
+
+1. New coins `SHALL` use the same number of decimals as `APT` for ease of
+   constant product calculations.
+1. Each Emojicoin's `symbol` field `SHALL` use hex-encoded UTF8.
+1. Each Emojicoin's `name` field `SHALL` use the format `A [+ B] emojicoin`
+   where `A` and optionally `B` are the official emoji name, for example
+   "Globe with Meridians emojicoin" or "Rocket + Rocket emojicoin".
```

Comment:
I thought we were going to do the emojis themselves in the `name` field? or am I misremembering

Just noting that the 32 byte restriction would leave a few combos out, especially the 3 combo ones since it adds 3 bytes for each ` + ` that joins the emojis

Personally I like the way the emojis look in the name field, even if it's like

> ☘️☘️☘️ emojicoin

Reply:
Nice catch here. I wrote this before we realized we couldn't fit the official name in the symbol field. So it's updated now

Reply:
Addressed in e2d23b84c766694c1ae1ae692e85404a4c447b9c